datalad/utils.py

```diff
@@ -1097,74 +993,6 @@ def  _wrap_line_profile(*args, **kwargs):
     return  _wrap_line_profile
 
 
-# unused in -core
-@optional_args
-def collect_method_callstats(func):
```

Comment:
I wouldn't delete this one because it's unused. To me it's meant to be used adhoc when assessing things locally, not something to decorate anything permanently with. There may be a better place to move it to, but deletion doesn't seem to be it.

Reply:
`datalad.tests.utils.py` sounds like a better place then.

How do we know that it works? ;-)

Reply:
ha -- and isn't it something you @mih wanted (to figure out what actually uses some function)? ;)
I would keep it around.  It did come handy in the original use case (183b0202723aed3b1e2fa32ab5c55833312434ec) and to say the truth I did forget about it. Indeed I do not expect it to be used in the main code UNLESS we actually decide to establish some optional (triggered e.g. by env var) collection of usage stats, and per se it is not really about helping tests machinery.  As for "how do we know" - I guess someone is yet to compose a test for it , and may be indeed meanwhile move such unused but potentially useful into something like `datalad._utils` which would disappear from the public API surface?

Reply:
> may be indeed meanwhile move such unused but potentially useful into something like datalad._utils which would disappear from the public API surface?

Agree with the reasoning and `datalad._utils` sounds better to me than `datalad.tests.utils`. And alternative may be to have a module for such debugging/analyzing under `datalad/tools`?

Reply:
ha - I was always wondering on what difference is between `utils` and `tools`.  I guess it could be it -- `utils` are for use in the codebase, and `tools` -- for internal use/debugging/etc.  Albeit easy to mix up, I think a clear docstring describing that  and stating that those interfaces might change without due DataLad version change should be be enough.